### PR TITLE
Fix build issues with recent Cython versions.

### PR DESCRIPTION
**Pull request by rainwoodman**
(And first time building on a M2 aarch64.
runtests passed. Yay)

### CODE REVIEW

**Comment by rainwoodman** [comment]
https://github.com/bccp/nbodykit/issues/684

File: Makefile [hunk 1]
```diff
@@ -20,7 +20,7 @@ main: main.c libmpsort-omp.a libradixsort.a
 	$(CC) $(CFLAGS) -o main $^
 main-mpi: main-mpi.c libmpsort-mpi.a libradixsort.a
 	$(MPICC) $(CFLAGS) -o main-mpi $^
-bench-mpi: bench-mpi.c libmpsort-mpi.a libradixsort.a mp-mpiu.h
```
Comment (sbird) [inline Makefile:23]: Do you mean to have this *and* the change below?

File: bench-mpi.c [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 #include <mpi.h>
 #include "mpsort.h"
-#include "mpiu.h"
+#include "mp-mpiu.h"
```
Comment (sbird) [inline bench-mpi.c:11]: This conflicts maybe with the change above?

**Comment by sbird** [comment]
I didn't know that aarch-64 required these exception specifiers! But this looks fine to me.

**Review by sbird — APPROVED** [review]
Ok!

**Comment by rainwoodman** [comment]
The exception specifiers was due to recent version of Cython. I think the most relevant bits was the gcc compiler gets more strict about including stddef and other header files.

**Pull request merged**